internal/cache/lru/lru.go

```diff
@@ -214,3 +218,49 @@ func (c *Cache) LookUp(key string) (value ValueType) {
 	// Return the value.
 	return e.Value.(entry).Value
 }
+
+// LookUpWithoutChangingOrder looks up previously-inserted value for a given key
+// without changing the order of entries in the cache. Return nil if no value
+// is present.
+func (c *Cache) LookUpWithoutChangingOrder(key string) (value ValueType) {
```

Comment:
Should we use `sync.RWMutex` to allow multiple readers together for this method?

Reply:
That's a good catch. Thanks !